### PR TITLE
Stanley development

### PR DESCRIPTION
**Pull request by stanleynwakamma**
Please review

### CODE REVIEW

File: src/main/webapp/index.html [hunk 1]
```diff
@@ -28,7 +28,8 @@
         <li><a href="/">Home</a></li>
       </ul>
     </nav>
-    <h1> This is Stanley's first change</h1>
+    <h1> This is Stanley's first change! Everthing here is pretty cool</h1>
+    <h2> Team 10 is the best Team ever </h2>
```
Comment (kyrabusser) [inline src/main/webapp/index.html:32]: I like this new subheading a lot, but it's not related to the messages changes. Could you undo the changes to index.html in this PR, and make a new branch with a different PR if you'd like to change this file?

File: src/main/java/com/google/codeu/data/Message.java [hunk 2]
```diff
@@ -18,14 +18,28 @@
 
 import java.util.UUID;
 
-/** A single message posted by a user. */
+/** The sting recipient helps to display messages received from another user. */
```
Comment (kyrabusser) [inline src/main/java/com/google/codeu/data/Message.java:21]: Good job adding comments! However, this comment is describing the entire class, which is still a message posted by a user, not just the new string you've added. Maybe change this back, or modify it to "A single message posted from a user to a recipient"?

File: src/main/java/com/google/codeu/data/Message.java [hunk 3]
```diff
@@ -18,14 +18,28 @@
...
 public class Message {
 
+  private String recipient;
```
Comment (kyrabusser) [inline src/main/java/com/google/codeu/data/Message.java:24]: This belongs together with the other private member variables (id, user, text, timestamp) on lines 37-40.

File: src/main/java/com/google/codeu/data/Message.java [hunk 4]
```diff
@@ -18,14 +18,28 @@
...
+  public Message(String user, String text, String recipient) {
+  this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient);
+}
+
+public Message(UUID id, String user, String text, long timestamp, String recipient) {
```
Comment (kyrabusser) [inline src/main/java/com/google/codeu/data/Message.java:29]: Here you've added two brand new constructors with the new recipient field we want. What we actually want is to *change* the constructors that already exist (lines 47 to 56).

**Review by hengelman — APPROVED** [review]
Looks good to me. In the future can you make sure the commit title or a comment explains exactly what function you are implementing?

**Review by kyrabusser — CHANGES_REQUESTED** [review]
Great job, Stanley! :tada: I'm excited to see this feature coming together. :smile: 

I've made some comments based on https://sites.google.com/codeustudents.com/spring-2019/weeks-1-2-familiarization-projects/direct-messages

Can you make these changes and then also test that `mvn appengine:devserver` works? You might need to combine multiple steps into a single PR if we want it to compile, but that's OK for this feature.

File: src/main/java/com/google/codeu/data/Message.java (hunk 2 above)
Comment (stanleynwakamma) [reply src/main/java/com/google/codeu/data/Message.java:21]: Thank you so much! I will implement the change

Comment (kyrabusser) [reply src/main/java/com/google/codeu/data/Message.java:21]: Awesome! Add more commits to your same branch and then run `git push` again so we can see them here. 👍

File: src/main/java/com/google/codeu/data/Message.java (hunk 4 above)
Comment (stanleynwakamma) [reply src/main/java/com/google/codeu/data/Message.java:29]: Resolved. Thank you 😄

File: src/main/webapp/index.html (hunk 1 above)
Comment (stanleynwakamma) [reply src/main/webapp/index.html:32]: Sure I'm on it right now

**Comment by stanleynwakamma** [comment]
Thank you @hengelman and @kyrabusser for all your thoughts and advice. I really appreciate it. This is my first time using GitHub, I'm really doing my best in learning how to use the platform more efficiently. 😄  Please pardon my many errors 😆  Thanks again everyone!

**Comment by kyrabusser** [comment]
So glad you're stretching and learning new things! One thing I noticed is that you marked this PR as closed--is that intentional, or do you want to reopen it?

**Comment by stanleynwakamma** [comment]
So I need some clarifications. Do I need to create a new pull request each time there's a conflict, like do I need to close the current PR and create a new PR with the implemented changes after I have resolved the conflict?
What is the most effective way to resolve conflicts on GitHub (rewriting the code) without having to go back to my text editor? 
Assuming I went back to my text editor to rewrite the code, how do I update it on the current conflicted PR without having to create a new PR from my console?
How do I make sure the code in my local computer is always up to date with the master on GitHub?
Thanks
@kyrabusser @hengelman @CharlesArday

**Comment by stanleynwakamma** [comment]
I know that's a lot of questions 🤣 Please pardon me. I'm just trying to get better 🙂

**Comment by kyrabusser** [comment]
Questions are great! Thanks for asking them and learning new things! :smile: 

Nope, you don't need to make a new PR--in fact, if you keep using the same one, then we can all see the history. As long as you're making the changes in new commits and pushing them up to your same branch, they'll show in the PR. But no worries about this for this one--we can just continue in #10 :+1: 

You _can_ edit code or resolve conflicts using the GitHub web interface, but I think I would recommend going back to your text editor instead. That way we're only editing in one place, and the flow is more consistent.

A pull request is asking to merge one branch into another, and a branch can be a whole series of many commits. That means you can add new commits to the same branch, then push them to GitHub, and they'll be part of the same PR.

To keep your local code up-to-date, use `git pull origin master` early and often! This will grab all the latest changes from GitHub and put them in your local code. If necessary, this will also create a merge commit, and if there are conflicts it will warn you about them and make you resolve them locally. See the CodeU site here: https://sites.google.com/codeustudents.com/spring-2019/codeu-toolbox/reference-guides/using-git#h.p_Ru9ON8xbx9fC

Hope that helps! Please always let me know when questions come up. :+1:
